Prox/Prox/Data/PlacesController.swift

```diff
@@ -126,8 +128,15 @@ class PlacesProvider {
         database.getPlaces(forLocation: location, withRadius: radius).upon { results in
             let places = results.flatMap { $0.successResult() }
             let placeCount = places.count
+
+            guard retriesLeft > 0 else {
+                self.isUpdating = false
+                self.delegate?.placesProviderDidTimeout(self)
```

Comment:
Either this needs to be called from the main thread, or the delegate needs to perform it's actions on the main thread once it's called as I'm getting a load of these warnings:

```
2016-11-25 09:54:52.235 Prox[31633:645351] This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   CoreFoundation                      0x000000010a6f634b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010a15721e objc_exception_throw + 48
	2   CoreFoundation                      0x000000010a75f265 +[NSException raise:format:] + 197
	3   Foundation                          0x0000000109e522b8 _AssertAutolayoutOnAllowedThreadsOnly + 180
	4   Foundation                          0x0000000109e52000 -[NSISEngine _optimizeWithoutRebuilding] + 61
	5   Foundation                          0x0000000109c79dfc -[NSISEngine optimize] + 91
	6   Foundation                          0x0000000109e4ff20 -[NSISEngine performPendingChangeNotifications] + 84
	7   UIKit                               0x000000010b979b16 -[UIView(Hierarchy) layoutBelowIfNeeded] + 265
	8   Prox                                0x0000000107e3f3a5 _TFC4Prox18LoadingOverlayView15fadeInMessagingfT_T_ + 53
	9   Prox                                0x0000000107e288e1 _TFC4Prox27PlaceCarouselViewControllerP33_E0A168B46C9580E142C881F80CE2EF0926showErrorMessageIfNoPlacesfT_T_ + 145
	10  Prox                                0x0000000107e2900c _TFC4Prox27PlaceCarouselViewController24placesProviderDidTimeoutfCS_14PlacesProviderT_ + 28
	11  Prox                                0x0000000107e29171 _TTWC4Prox27PlaceCarouselViewControllerS_22PlacesProviderDelegateS_FS1_24placesProviderDidTimeoutfCS_14PlacesProviderT_ + 49
	12  Prox                                0x0000000107e468a0 _TFFC4Prox14PlacesProviderP33_34802027475AD4230BD0432332D06C1516retryQueryPlacesFT8locationCSo10CLLocation10withRadiusSd11retriesLeftSi9lastCountSi_T_U_FGSaGCS_14DatabaseResultCS_5Place__T_ + 2096
	13  Prox                                0x0000000107dc8638 _TTRXFo_oGSaGC4Prox14DatabaseResultCS_5Place____XFo_iGSaGS0_S1_____ + 24
	14  Deferred                            0x00000001098e1735 _TFFC8Deferred8DeferredP33_8471EEB727F82F3A241B76FBEA842E6D6notifyFT5flagsV8Dispatch21DispatchWorkItemFlags4uponCSo13DispatchQueue7executeFxT__T_U_FT_T_ + 469
	15  Deferred                            0x00000001098e1b27 _TTRgrXFo___XFdCb___ + 39
	16  libdispatch.dylib                   0x000000010db2b278 _dispatch_block_async_invoke_and_release + 75
	17  libdispatch.dylib                   0x000000010db500cd _dispatch_client_callout + 8
	18  libdispatch.dylib                   0x000000010db31691 _dispatch_root_queue_drain + 1450
	19  libdispatch.dylib                   0x000000010db3108b _dispatch_worker_thread3 + 123
	20  libsystem_pthread.dylib             0x000000010def94de _pthread_wqthread + 1129
	21  libsystem_pthread.dylib             0x000000010def7341 start_wqthread + 13
```